### DISCUSSION TITLE
Line graph of a directed graph

### DISCUSSION THREAD

**Discussion by vymao** [networkx/networkx#4387] [Q&A] [answered]
Does networkx follow the convention [set forth by Wolfram](https://mathworld.wolfram.com/LineGraph.html) for the line graph of a directed graph? 

> The line graph of a directed graph G is the directed graph L(G) whose vertex set corresponds to the arc set of G and having an arc directed from an edge e_1 to an edge e_2 if in G, the head of e_1 meets the tail of e_2

**Comment by dschult** [comment]
What does Wolfram mean by "meets the tail of"?

I hope our docs are better:  https://networkx.org/documentation/stable//reference/generated/networkx.generators.line.line_graph.html

**Comment by vymao** [comment]
By tail of, Wolfram means the source of a directed edge (and so head means target node). I am interpreting this based on their diagram. 

I read the docs: 
> nodes are adjacent exactly when the edges they represent form a directed path of length two

But I am not sure what "adjacent" entails. Are the nodes of the line graph still connected by directed edges? If so, how are these directions determined?

**Comment by jarrodmillman** [comment, marked as answer]
@vymao Maybe this picture will help you:
```
import networkx as nx
import matplotlib.pyplot as plt

G = nx.DiGraph()
G.add_edges_from([('A', 'B'),  ('B', 'C'), ('C', 'A')])

H = nx.line_graph(G)

plt.subplot(211)
plt.title("G")
nx.draw_networkx(G)

plt.subplot(212)
plt.title("line_graph(G_")
nx.draw_networkx(H)

plt.show()
```

![Figure_1](https://user-images.githubusercontent.com/123428/100146757-3c116880-2e4f-11eb-8ece-6684825da79d.png)
